drivers/flash/flash_stm32_ospi.c

```diff
@@ -942,7 +943,11 @@ static int flash_stm32_ospi_write(const struct device *dev, off_t addr,
 	/* page program for STR or DTR mode */
 	OSPI_RegularCmdTypeDef cmd_pp = ospi_prepare_cmd(dev_cfg->data_mode, dev_cfg->data_rate);
 
-	cmd_pp.Instruction = dev_data->write_opcode;
+	/* using 32bits address also in SPI/STR mode */
+	cmd_pp.Instruction = (dev_cfg->data_mode == OSPI_OPI_MODE)
+			? SPI_NOR_OCMD_PAGE_PRG
+			: SPI_NOR_CMD_PP_4B;
+	/* Do not rely on SFDP:BFP write instruction */
```

Comment:
why hardcoding write opcode? That is similar to the read opcode situation... That breaks the driver for some ICs.
If SFDP alone is a problem, then the driver shall use the SFDP opcodes but allow users to override that in DTS.

The SFDP is helpful as never before due to semiconductor shortages...

Reply:
In case of SPi/STR the instruction must be SPI_NOR_CMD_PP_4B, else 
```
Data read does not match data written!!                                         
000ff000 wrote 55 read ff MISMATCH                                              
000ff001 wrote aa read ff MISMATCH                                              
000ff002 wrote 66 read ff MISMATCH                                              
000ff003 wrote 99 read ff MISMATCH   
```

Consequently setting  `cmd_pp.Instruction = dev_data->write_opcode` requires a 
```
case OSPI_SPI_MODE:
			data->write_opcode = SPI_NOR_CMD_PP_4B;
			break;
```

in the `spi_nor_process_bfp` function

Reply:
> In case of **SPI/STR**, the instruction _must_ be SPI_NOR_CMD_PP_4B, else
> 
> ```
> Data read does not match data written!!                                         
> 000ff000 wrote 55 read ff MISMATCH                                              
> 000ff001 wrote aa read ff MISMATCH                                              
> 000ff002 wrote 66 read ff MISMATCH                                              
> 000ff003 wrote 99 read ff MISMATCH   
> ```
> 
> Consequently setting `cmd_pp.Instruction = dev_data->write_opcode` requires a
> 
> ```
> case OSPI_SPI_MODE:
> 			data->write_opcode = SPI_NOR_CMD_PP_4B;
> 			break;
> ```
> 
> in the `spi_nor_process_bfp` function

@FRASTM, please enable `four-byte-opcodes` in the DTS for your flash, which shall use a 4-byte opcode (`SPI_NOR_CMD_PP_4B`) in the case of `OSPI_SPI_MODE`.

Reply:
ok done for b_u585i_iot02a and stm32l562e_dk